audio/out/ao_coreaudio_utils.c

```diff
@@ -455,28 +455,36 @@ int64_t ca_get_device_latency_ns(struct ao *ao, AudioDeviceID device)
     return MP_TIME_S_TO_NS(latency_frames / sample_rate);
 }
 
+struct cb_sem {
+    mp_mutex mutex;
+    mp_cond cond;
+};
+
 static OSStatus ca_change_format_listener(
     AudioObjectID object, uint32_t n_addresses,
     const AudioObjectPropertyAddress addresses[],
     void *data)
 {
-    mp_sem_t *sem = data;
-    mp_sem_post(sem);
+    struct cb_sem *sem = data;
+    mp_mutex_lock(&sem->mutex);
+    mp_cond_broadcast(&sem->cond);
+    mp_mutex_unlock(&sem->mutex);
     return noErr;
 }
 
 bool ca_change_physical_format_sync(struct ao *ao, AudioStreamID stream,
                                     AudioStreamBasicDescription change_format)
 {
+    static struct cb_sem sem = {
```

Comment:
Do you want to have it static? I think it would be better to keep it in `struct ao`

Reply:
It is only used in this single function. I don't think it is worth bloating `struct ao`.

Reply:
Now we are bloating global namespace instead.

Reply:
It only got initialized on the first time ca_change_physical_format_sync is called.

Maybe we can put it in `ao->priv`.

Reply:
Modified in c3701bd